src/vs/workbench/node/extensionPoints.ts

```diff
@@ -81,22 +83,37 @@ export class PluginScanner {
 	/**
 	 * Scan a list of extensions defined in `absoluteFolderPath`
 	 */
-	public static scanPlugins(version: string, collector: IPluginsMessageCollector, absoluteFolderPath:string, isBuiltin:boolean): TPromise<IPluginDescription[]> {
-		return pfs.readDirsInDir(absoluteFolderPath).then((folders) => {
-			return TPromise.join(
-				folders.map((folder) => this.scanPlugin(version, collector, paths.join(absoluteFolderPath, folder), isBuiltin))
-			);
-		}, (err) => {
-			collector.error(absoluteFolderPath, err);
-			return [];
-		}).then((results) => results.filter(item => (item !== null)));
+	public static scanPlugins(
+		version: string,
+		collector: IPluginsMessageCollector,
+		absoluteFolderPath:string,
+		isBuiltin:boolean
+	) : TPromise<IPluginDescription[]>
+	{
+		return pfs.readDirsInDir(absoluteFolderPath)
+			.then(folders => TPromise.join(folders.map(f => this.scanPlugin(version, collector, paths.join(absoluteFolderPath, f), isBuiltin))))
+			.then(plugins => plugins.filter(item => item !== null))
+			.then(plugins => {
+				const pluginsById = values(groupBy(plugins, p => p.id));
+				return pluginsById.map(p => p.sort((a, b) => semver.rcompare(a.version, b.version))[0]);
```

Comment:
`semver.rcompare(a.version, b.version)` throws if `a.version` or `b.version` are undefined.

Effectively, this change makes `version` a required field in `package.json`. If this is the intent, then the validation of `package.json` must reflect this. If the version field is required, then `pluginsRegistry.ts` > `isValidPluginDescription` must be updated to check the field exists and **is parsable by semver**.

The following `package.json` breaks all extension loading:
```
{
	"name": "test",
	"publisher": "alex",
	"engines": {
		"vscode": "0.10.x"
	}
}
```

Just place it twice in the extensions folder with two different folder names to repro.